raw_data_gen.py

```diff
@@ -0,0 +1,140 @@
+def get_size_hints(model: str, kernel: str, src: str) -> list:
+    startpos = src.find("size_hints=[")
+    assert startpos != -1
+    endpos = src.find("]", startpos)
+    return parse_list_of_numbers(src[startpos + len("size_hints=[") : endpos])
+    
+    
+for DB_DIR in DB_DIRS:
+    for model in sorted(listdir(DB_DIR)):
+        model_path = join(DB_DIR, model)
+        if not isdir(model_path):
+            continue
+
+        for kernel in sorted(listdir(model_path)):
+            kernel_path = join(model_path, kernel)
+            if not isdir(kernel_path):
+                continue
+
+            for py in listdir(kernel_path):
+                py_path = join(kernel_path, py)
+                if not py.endswith(".py"):
+                    continue
+
+                with open(py_path, "r") as file:
+                    src = file.read()
+                    if "Original ATen:" in src:
+                        continue
+
+                kernel_name = py[:-3]
+                log_path = join(kernel_path, kernel_name + ".log")
+                pkl_path = join(kernel_path, py + ".pkl")
+                all_config_path = join(kernel_path, kernel_name + ".all_config")
+
+                # Get the kernel category
+                # Some kernels are just eliminated by the compiler
+                kernel_category = get_kernel_category(model, kernel_name, src)
+                if kernel_category is None:
+                    continue
+                
+                # Sanity check
+                if os.path.exists(log_path) and os.path.exists(pkl_path) and os.path.exists(all_config_path):
+                    seen_kernels.add(kernel_name)
+                elif not os.path.exists(log_path) or not os.path.exists(pkl_path) or not os.path.exists(all_config_path):
+                    if not kernel_name in seen_kernels:
+                        logger.warning(f"Missing {model}, {kernel_name}")
+                    continue
+                else:
+                    logger.warning(f"Incomplete {model}, {kernel_name}")
+                    continue
+
+                # Get the number of loops
+                if kernel_category is KernelCategory.REDUCTION:
+                    num_of_loops = get_number_of_loops(model, kernel_name, src)
+                else:
+                    num_of_loops = 0
+
+                kernel_counter = kernel_counter + 1
+                
+                # Map the ops to numbers
+                (reads, writes, total_bytes), nodes, node_read_writes, src_code = tuple(pickle.load(open(pkl_path, "rb")))
+                op_counts = node_read_writes.op_counts
+                op_bag = dict()
+                for op in op_counts.keys():
+                    if op in op_dict:
+                        op_bag[op_dict[op]] = op_counts[op]
+                    else:
+                        op_dict[op] = op_counter
+                        op_bag[op_counter] = op_counts[op]
+                        op_counter = op_counter + 1
```

Comment:
Here the ids assigned to ops willy vary from run to run based on ordering.  We should create a fixed vocab.